src/common/imgix-js-core-wrapper.ts

```diff
@@ -35,35 +26,28 @@ export const createURLBuilderFn = <T extends 'buildURL' | 'buildSrcSet'>(
   fn: T,
 ) => (options?: Parameters<typeof createImgixClient>[0]) => (
   ...args: Parameters<InstanceType<typeof ImgixClient>[T]>
-): string =>
-  pipe(
-    Do(E.either)
-      .bindL('urlParts', () => {
-        if (options?.domain) {
-          return E.right({
-            domain: options?.domain,
-            path: args[0],
-          });
-        }
-        return sequenceS(E.either)({
-          domain: parseHostE(args[0]),
-          path: parsePathE(args[0]),
-        });
-      })
-      .bindL('client', ({ urlParts: { domain } }) =>
-        createImgixClient({
-          ixlib: 'gatsbyFP',
-          ...options,
-          domain,
-        }),
-      )
-      .return(({ client, urlParts: { path } }) =>
-        client[fn](path, ...args.slice(1)),
-      ),
-    E.getOrElse<Error, string>((err) => {
-      throw err;
-    }),
-  );
+): string => {
+  const urlParts = (() => {
+    if (options?.domain) {
+      return {
+        domain: options?.domain,
+        path: args[0],
+      };
+    }
+    return {
+      domain: parseHost(args[0]),
+      path: parsePath(args[0]),
+    };
+  })();
+
+  const client = createImgixClient({
+    ixlib: 'gatsbyFP',
+    ...options,
+    domain: urlParts.domain,
+  });
+
+  return client[fn](urlParts.path, ...args.slice(1));
```

Comment:
I feel stupid, but why are we removing the first argument here again?

Reply:
Do you mean the `...args.slice(1)` part? If so, that is because that argument is the `path`, which we are passing in with `urlPaths.path`. Does this answer your question?